src/chocolatey/infrastructure.app/commands/ChocolateySearchCommand.cs

```diff
@@ -0,0 +1,260 @@
+                 .Add("id-starts-with",
+                     "IdStartsWith - Only return packages where the id starts with the search filter. Available in 0.9.10+.",
+                     option => configuration.ListCommand.IdStartsWith = option != null)
+                 .Add("order-by-popularity",
+                     "OrderByPopularity - Sort by package results by popularity. Available in 0.9.10+.",
+                     option => configuration.ListCommand.OrderByPopularity = option != null)
+                 .Add("approved-only",
+                    "ApprovedOnly - Only return approved packages - this option will filter out results not from the community repository. Available in 0.9.10+.",
+                     option => configuration.ListCommand.ApprovedOnly = option != null)
+                 .Add("download-cache|download-cache-only",
+                     "DownloadCacheAvailable - Only return packages that have a download cache available - this option will filter out results not from the community repository. Available in 0.9.10+.",
+                     option => configuration.ListCommand.DownloadCacheAvailable = option != null)
+                 .Add("not-broken",
+                     "NotBroken - Only return packages that are not failing testing - this option only filters out failing results from the community feed. It will not filter against other sources. Available in 0.9.10+.",
+                     option => configuration.ListCommand.NotBroken = option != null)
+                  .Add("detail|detailed",
+                     "Detailed - Alias for verbose. Available in 0.9.10+.",
+                     option => configuration.Verbose = option != null)
+                  .Add("disable-repository-optimizations|disable-package-repository-optimizations",
+                    "Disable Package Repository Optimizations - Do not use optimizations for reducing bandwidth with repository queries during package install/upgrade/outdated operations. Should not generally be used, unless a repository needs to support older methods of query. When disabled, this makes queries similar to the way they were done in Chocolatey v0.10.11 and before. Overrides the default feature '{0}' set to '{1}'. Available in 0.10.14+.".format_with
+                        (ApplicationParameters.Features.UsePackageRepositoryOptimizations, configuration.Features.UsePackageRepositoryOptimizations.to_string()),
+                    option =>
+                    {
+                        if (option != null)
+                        {
+                            configuration.Features.UsePackageRepositoryOptimizations = false;
+                        }
+                    })
+                ;
+        }
+
+        public virtual void handle_additional_argument_parsing(IList<string> unparsedArguments, ChocolateyConfiguration configuration)
+        {
+            configuration.Input = string.Join(" ", unparsedArguments);

```

Comment:
Same comment as above; why would we allow searching for package names that can't be used with `install` / etc?

I guess _maybe_ you'd use it to search for stuff in a description... That really sounds like the edgiest of edge cases, and even a lot of modern search engines don't handle that kind of case super well. I'd be curious to se if trying to search for something in a form like `-query term` would be able to find anything useful on CCR or in most other nuget feed implementations to really warrant bothering to handle that kind of thing.

Reply:
Actually, ignore this; this is the same as the current list/search behaviour, no point changing it right now.